app/web_framework/cluster_master_application.py

```diff
@@ -33,6 +39,7 @@ def __init__(self, cluster_master):
         root_route = \
             RouteNode(r'/', _RootHandler).add_children([
                 RouteNode(r'v1', _APIVersionOneHandler).add_children([
+                    RouteNode(r'metrics', _MetricsHandler),
```

Comment:
Maybe we could just leave the endpoint active in either case, and if it's turned off just have it return an error code.